Sources/BraintreeDropIn/BTPaymentSelectionViewController.m

```diff
@@ -64,130 +57,43 @@ - (instancetype)initWithAPIClient:(BTAPIClient *)apiClient request:(BTDropInRequ
-                                                                      options:0
-                                                                      metrics:[BTUIKAppearance metrics]
-                                                                        views:viewBindings]];
-
-    NSLayoutConstraint *heightConstraint;
-    self.vaultedPaymentsHeader = [self sectionHeaderLabelWithString:BTUIKLocalizedString(RECENT_LABEL)];
-    self.vaultedPaymentsHeader.translatesAutoresizingMaskIntoConstraints = NO;
-
-    self.vaultedPaymentsLabelContainerStackView = [self newStackView];
-    self.vaultedPaymentsLabelContainerStackView.axis  = UILayoutConstraintAxisHorizontal;
-    self.vaultedPaymentsLabelContainerStackView.layoutMargins = UIEdgeInsetsMake(0, [BTUIKAppearance horizontalFormContentPadding], 0, [BTUIKAppearance horizontalFormContentPadding]);
-    self.vaultedPaymentsLabelContainerStackView.layoutMarginsRelativeArrangement = true;
-
-    [self.vaultedPaymentsLabelContainerStackView addArrangedSubview:self.vaultedPaymentsHeader];
-
-    self.vaultedPaymentsEditButton = [UIButton new];
-    self.vaultedPaymentsEditButton.hidden = YES;
-
-    [self.vaultedPaymentsEditButton setTitle:BTUIKLocalizedString(EDIT_ACTION) forState:UIControlStateNormal];
-    [self.vaultedPaymentsEditButton setTitleColor:[BTUIKAppearance sharedInstance].tintColor forState:UIControlStateNormal];
-    [self.vaultedPaymentsEditButton setTitleColor:[BTUIKAppearance sharedInstance].highlightedTintColor forState:UIControlStateHighlighted];
-    [self.vaultedPaymentsEditButton setTitleColor:[BTUIKAppearance sharedInstance].disabledColor forState:UIControlStateDisabled];
-    self.vaultedPaymentsEditButton.titleLabel.font = [BTUIKAppearance sharedInstance].staticBodyFont;
-
-    [self.vaultedPaymentsEditButton sizeToFit];
-    [self.vaultedPaymentsEditButton layoutIfNeeded];
-    [self.vaultedPaymentsEditButton addTarget:self action:@selector(vaultedPaymentsEditButtonPressed) forControlEvents:UIControlEventTouchUpInside];
-    [self.vaultedPaymentsLabelContainerStackView addArrangedSubview:self.vaultedPaymentsEditButton];
-
-    [self.stackView addArrangedSubview:self.vaultedPaymentsLabelContainerStackView];
+    self.navigationItem.leftBarButtonItem = [[BTUIKBarButtonItem alloc] initWithTitle:BTUIKLocalizedString(CANCEL_ACTION)
+                                                                                style:UIBarButtonItemStylePlain
+                                                                               target:self
+                                                                               action:@selector(cancelButtonPressed:)];
+
+    UILabel *titleLabel = [BTUIKAppearance styledNavigationTitleLabel];
+    titleLabel.text = BTUIKLocalizedString(SELECT_PAYMENT_LABEL);
+    titleLabel.textAlignment = NSTextAlignmentCenter;
+    titleLabel.numberOfLines = 2;
```

Comment:
nitpick: If we set this to zero does it work with dynamic type? It's unlikely that it'll wrap to 3 lines but it couldn't hurt to have that insurance.

Reply:
Hmm... I think it would get cut off if it went to 3 lines because the navigation bar doesn't scale vertically. The text in the nav bar doesn't scale even with dynamic font, so this is just for small phones and languages where the text might need to be 2 lines. After two lines, I think we should just truncate (although hopefully that doesn't happen).